### DISCUSSION TITLE
Any idea why my String fields are all returning 255 characters?

### DISCUSSION THREAD

**Discussion by rootedsoftware** [prisma/prisma#3198] [Q&A] [answered]
I'm creating a document in the database (through Prisma to PostgresSQL). I'm then simply logging out the returned object and all of the String types are coming back with 255 character lengths. 

I can't even paste the results here because GitHub is removing the white space, but I've attached a screenshot of it and I've checked the length of the field when it is logged to the console and it is 255.

<img width="571" alt="image" src="https://user-images.githubusercontent.com/3142974/89250313-1b8c7c80-d5da-11ea-9838-3cca7a0c16cd.png">

**Comment by rootedsoftware** [comment, marked as answer]
I just figured out what I did wrong. I was using the guide https://www.prisma.io/docs/getting-started/setup-prisma/start-from-scratch-sql-node-postgres#create-database-tables-with-sql. This guide suggests using VARCHAR(255) however, the type mapping for PostgreSQL here https://www.prisma.io/docs/reference/database-connectors/postgresql#type-mapping-between-postgresql-to-prisma-schema shows that I should be using TEXT instead.

**Reply by janpio** [reply]
Those things sound unrelated. If you get back different data than what you have in the database, this is a clear bug. Please create an issue for that and include both your schema file and your SQL schema from the database (that should include `varchar(255)`.

**Comment by SH20RAJ** [comment]
To resolve the issue with string fields returning 255 characters in Prisma, ensure you use the `TEXT` type for PostgreSQL instead of `VARCHAR(255)` or blank string. Here's a sample schema change:

**Before:**
```prisma
model User {
  id    Int     @id @default(autoincrement())
  name  String  @db.VarChar(255)
}
```

**After:**
```prisma
model User {
  id    Int     @id @default(autoincrement())
  name  String  @db.Text
}
```

This change will ensure that the string fields do not get truncated to 255 characters.
> @db.Text: This attribute tells Prisma to use a text type in the database that can store large amounts of text (e.g., TEXT in PostgreSQL or MySQL, which can store up to 1GB of text).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
